common/meta/grpc/src/meta_client_on_kv_impl.rs

```diff
@@ -662,9 +664,66 @@ impl MetaApi for MetaClientOnKV {
 
     async fn upsert_table_option(
         &self,
-        _req: UpsertTableOptionReq,
+        req: UpsertTableOptionReq,
     ) -> Result<UpsertTableOptionReply, MetaError> {
-        todo!()
+        let tbid = TableId {
+            table_id: req.table_id,
+        };
+        let req_seq = req.seq;
+
+        loop {
+            let (tb_meta_seq, table_meta): (_, Option<TableMeta>) =
+                self.get_struct_value(&tbid).await?;
+
+            tracing::debug!(ident = display(&tbid), "upsert_table_option");
+
+            if req_seq.match_seq(tb_meta_seq).is_err() {
+                let err = AppError::from(TableVersionMismatched::new(
+                    req.table_id,
+                    req.seq,
+                    tb_meta_seq,
+                    "upsert_table_option",
+                ));
+
+                return Err(MetaError::AppError(err));
+            }
+            let mut table_meta = table_meta.unwrap();
```

Comment:
`table_meta` can be None, if the returned `tb_meta_seq` is 0.

Reply:
ok